src/System.Management.Automation/engine/runtime/Operations/MiscOps.cs

```diff
@@ -1164,26 +1164,51 @@ internal Pipe GetRedirectionPipe(ExecutionContext context, PipelineProcessor par
                 return new Pipe { NullPipe = true };
             }
 
-            CommandProcessorBase commandProcessor = context.CreateCommand("out-file", false);
-            Diagnostics.Assert(commandProcessor != null, "CreateCommand returned null");
-
-            // Previously, we mandated Unicode encoding here
-            // Now, We can take what ever has been set if PSDefaultParameterValues
-            // Unicode is still the default, but now may be overridden
+            // determine whether we're trying to set a variable by inspecting the file path
+            // if we can determine that it's a variable, we'll use Set-Variable rather than Out-File
+            ProviderInfo p;
+            PSDriveInfo d;
+            CommandProcessorBase commandProcessor;
+            var name = context.SessionState.Path.GetUnresolvedProviderPathFromPSPath(File, out p, out d);
 
-            var cpi = CommandParameterInternal.CreateParameterWithArgument(
-                /*parameterAst*/null, "Filepath", "-Filepath:",
-                /*argumentAst*/null, File,
-                false);
-            commandProcessor.AddParameter(cpi);
+            if (ExperimentalFeature.IsEnabled("PSRedirectToVariable") && p != null && p.NameEquals(context.ProviderNames.Variable))
```

Comment:
Should the name compare be case insensitive?

Reply:
> Should the name compare be case insensitive?

NameEquals is caseinsensitive.

Reply:
> p.NameEquals(context.ProviderNames.Variable)

Hmm, as a script developer now I ask why can I redirect only to file or variable? I'd expect the redirection works for other providers too. Aren't providers built to support universal behavior? I want `Get-Time > Env:\DATETIME`. I suppose this question will arise immediately in feedbacks.

Reply:
-